src/common/database.c

```diff
@@ -2998,6 +2998,9 @@ gboolean _ask_for_maintenance(const gboolean has_gui, const gboolean closing_tim
 
 void dt_database_maybe_maintenance(const struct dt_database_t *db, const gboolean has_gui, const gboolean closing_time)
 {
+  if(!g_strcmp0(db->dbfilename_data, ":memory:") || !g_strcmp0(db->dbfilename_library, ":memory:"))
```

Comment:
Since the new code is the same for both `dt_database_maybe_maintenance` and `dt_database_optimize`, I would make a unique function to be called in both places.

Reply:
It's an `if`statement. I could change it to `if(is_db_memory(db)) return;` would that be better?

Reply:
Exactly!